internal/pkg/manifest/env.go

```diff
@@ -415,6 +416,70 @@ func (cfg *EnvironmentHTTPConfig) loadLBConfig(env *config.CustomizeEnv) {
 type PublicHTTPConfig struct {
 	SecurityGroupConfig ALBSecurityGroupsConfig `yaml:"security_groups,omitempty"`
 	Certificates        []string                `yaml:"certificates,omitempty"`
+	ELBAccessLogs       ELBAccessLogsArgsOrBool `yaml:"access_logs,omitempty"`
+}
+
+// ELBAccessLogsArgsOrBool is a custom type which supports unmarshaling yaml which
+// can either be of type bool or type ELBAccessLogsArgs.
+type ELBAccessLogsArgsOrBool struct {
+	EnableAccessLogs  *bool
+	ELBAccessLogsArgs ELBAccessLogsArgs
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the ELBAccessLogsArgsOrBool
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v3) interface.
+func (al *ELBAccessLogsArgsOrBool) UnmarshalYAML(value *yaml.Node) error {
+	if err := value.Decode(&al.ELBAccessLogsArgs); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if !al.ELBAccessLogsArgs.isEmpty() {
+		// Unmarshaled successfully to al.AccessLogsArgs, reset al.EnableAccessLogs, and return.
+		al.EnableAccessLogs = nil
+		return nil
+	}
+
+	if err := value.Decode(&al.EnableAccessLogs); err != nil {
+		return errUnmarshalELBAccessLogs
+	}
+	return nil
+}
+
+// AccessLogsConfig holds the access logs configuration.
+// These options are specifiable under the "access_logs" field.
+type ELBAccessLogsArgs struct {
+	BucketName   *string `yaml:"bucket_name,omitempty"`
+	BucketPrefix *string `yaml:"bucket_prefix,omitempty"`
+}
+
+func (al *ELBAccessLogsArgs) isEmpty() bool {
+	return al.BucketName == nil && al.BucketPrefix == nil
+}
+
+func (al *ELBAccessLogsArgsOrBool) isEmpty() bool {
+	if al.EnableAccessLogs != nil {
+		return false
+	}
+	return al.ELBAccessLogsArgs.isEmpty()
+}
+
+// AccessLogs returns the access logs config if the user has set any values.
+// If there is no access logs settings, then returns nil and false.
+func (cfg *EnvironmentConfig) ELBAccessLogs() (*ELBAccessLogsArgs, bool) {
```

Comment:
what do you think of
```
func (cfg *EnvironmentConfig) ELBAccessLogs() (*template.ELBAccessLogs, bool) 
```

instead? I feel like this function and the one in `transformer.go` is sort of duplicated

Reply:
I'm a fan of keeping it as is and not return `template` types from the manifest pkg.
Ideally the `manifest` configuration can be transformed into multiple output types: CFN go templates, CDK code, Terraform (whatever we want) so it should return its own types.

All these convert functions would move to the `cfntemplate` pkg, and in unit tests we would ensure that `template.Parse` is called with the right arguments:

![template-dependencies](https://user-images.githubusercontent.com/879348/183473477-8f6905e1-0c5e-4961-81e1-d3d6f05ca609.png)